### PR TITLE
Delete old unused files

### PR DESCRIPTION
**Pull request by cmnrd**
This deletes the `.obsolete`, `.preliminary` and `.less-developed` subdirectories. As far as I can see, the content of these files is quite old and not up to date. I think it is safe to delete them. Please double-check.

### CODE REVIEW

**Comment by cmnrd** [comment]
Pinging @tanneberger as `docs/.less-developed/future-proof-package-and-import-system.mdx` and `docs/.preliminary/import-system.mdx` contain information that could be useful for the discussion on import mechanisms and syntax.

**Review by edwardalee — APPROVED** [review]
LGTM.  Let's let Martin Schoeberl capture what is useful from the Patmos file.  I've made a local record or the related-work file, which I do think is useful but probably doesn't belong on our website.

**Comment by axmmisaka** [comment]
> I think it is safe to delete them.

I can confirm these pages are not used/referenced anywhere

**Comment by tanneberger** [comment]
@cmnrd I made a backup of master. So these files don't get lost.

**Comment by axmmisaka** [comment]
> @cmnrd I made a backup of master. So these files don't get lost.

Sidenote, git is created for this purpose, we can just `git checkout` files which we removed

**Comment by tanneberger** [comment]
> > @cmnrd I made a backup of master. So these files don't get lost.
> 
> Sidenote, git is created for this purpose, we can just `git checkout` files which we removed

That's what I did locally ;D

**Comment by cmnrd** [comment]
@schoeberl Is it Ok to merge this?

**Review by axmmisaka — APPROVED** [review]
From the website point of view LGTM

File: docs/.preliminary/reactors-on-patmos.mdx
Comment (lhstrh) [inline docs/.preliminary/reactors-on-patmos.mdx:1]: This text is now available here: https://github.com/lf-lang/lf-lang.github.io/pull/245

File: docs/.less-developed/future-proof-package-and-import-system.mdx
Comment (lhstrh) [inline docs/.less-developed/future-proof-package-and-import-system.mdx:1]: I will review this and might submit a version of this as an RFC.

**Review by lhstrh — APPROVED** [review]

**Pull request merged**